### PR TITLE
modRest refactoring

### PR DESCRIPTION
**Pull request by Jako**
### What does it do?

Some code refactoring
### Why is it needed?
- The return values don't match the PHPDoc definitions and the usage was different too
- Wrong phpDoc type in recursive method call
- Removed unused variables
- Updated property descriptions
- Method names are case sensitive

### CODE REVIEW

File: core/model/modx/rest/modrest.class.php [hunk 1]
```diff
@@ -473,7 +473,7 @@ public function toXml($parameters,$rootNode) {
      * @param array $parameters
      */
     protected function _populateXmlDoc(&$doc, &$node, &$parameters) {
-        /** @var $val DOMNode */
+        /** @var DOMnode/array $val */
```
Comment (Mark-H) [inline core/model/modx/rest/modrest.class.php]: I don't think that's the right format for phpdoc is it? Isn't that supposed to be `DOMnode[]`?

**Review by Mark-H — COMMENTED** [review]
Could you elaborate on the reasoning behind changing from returning values to returning nothing? Sounds like a breaking change and I don't immediately see the benefit, so some more thoughts on what these changes accomplish would be great.

File: core/model/modx/rest/modrest.class.php (hunk 1 above)
Comment (Jako) [inline core/model/modx/rest/modrest.class.php]: Maybe `DOMnode[]/DOMnode`

**Comment by Jako** [comment]
The main reason for the change was that $this->success and $this->failure have no return values. The other methods use those return values. So the return was null every time (if the success and failure methods are not overridden).

**Comment by pixelchutes** [comment]
Thanks @Jako, was curious about the reasoning there.

File: core/model/modx/rest/modrest.class.php (hunk 1 above)
Comment (Alroniks) [inline core/model/modx/rest/modrest.class.php]: Alternatives should be separated using `|` instead of `/`. So correct way `DOMnode|array`

Comment (Jako) [inline core/model/modx/rest/modrest.class.php]: $parameters has now the right phpdoc type. The type warnings are away now.

**Comment by Mark-H** [comment]
Thanks @Jako!

**Pull request merged**